internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -225,6 +227,19 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 		}
 	}
 
+	// Set HTTPS Ignition URL
+	if *cluster.Status == models.ClusterStatusFinalizing {
```

Comment:
We need to do it once, and only if ignitionEndpoint isn't already set.
There is no reason to keep getting the MCS cert from the spoke if the ignitionEndpoint is set.